pkg/occlient/occlient.go

```diff
@@ -2123,6 +2141,39 @@ func (c *Client) Delete(labels map[string]string) error {
 
 }
 
+// WaitForComponentDeletion waits for component to be deleted
+func (c *Client) WaitForComponentDeletion(selector string) error {
+
+	glog.V(4).Infof("Waiting for component to get deleted")
+
+	watcher, err := c.appsClient.DeploymentConfigs(c.Namespace).Watch(metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+	eventCh := watcher.ResultChan()
+
+	for {
```

Comment:
Seems that all the case statements in the select are terminating which means we might not need the for loop?

Reply:
yes even i thought so, and removed the for loop but it did not work, since we get events on `eventCh` other than `error` `deleted`, case events gets selected and executes none of the if conditions, returns from the function.

Reply:
Gotcha, maybe an idea would be to just v4 log those events for debugging purposes?

Reply:
yup, sure will update for that.

Reply:
maybe lets do that in a follow up?

Reply:
created a follow up https://github.com/openshift/odo/issues/2879